elasticdl/python/worker/worker.py

```diff
@@ -202,6 +188,16 @@ def _init_from_args(self, args):
             checkpoint_path=args.checkpoint_dir,
         )
 
+        self._allreduce_trainer = None
+        if (
+            self._distribution_strategy == DistributionStrategy.ALLREDUCE
+            and args.num_workers > 1
+        ):
+            self._allreduce_trainer = AllReduceTrainer(
+                self._mc, self._model, self._loss, self._opt
+            )
+            self._allreduce_trainer.set_horovod_env(args.master_addr)
```

Comment:
master_addr can be an argument of `AllReduceTrainer` init, and `set_horovod_env` in init.